### DISCUSSION TITLE
On upgrade to version 7 the execution order of the methods is changed and priority is ignored.

### DISCUSSION THREAD

**Discussion by doctor-beat** [cbeust/testng#3061] [Q&A]
Hi, i am trying to upgrade testng from 6.x to 7.x. But when doing so the execution order of the test-methods is changed and the priority that is set on the method is ignored (or i should say is only evaluated inside the class, not across the suite).

I have written an IMethodInterceptor to restore ordering but the result is simply ignored by TestNg.

So, how can i restore the old order of execution in TestNg 7.x?

**Comment by krmahadevan** [comment]
Please share a sample gradle/maven project that can be used to reproduce the issue.
Also share what is the expected and actual output so that we can try and figure out what went wrong.

**Comment by doctor-beat** [comment]
https://github.com/doctor-beat/TestngPrioDemo

I found that the issue described only shows when you add a '-Dtest=*Something' to the 'mvn clean test' command
